projects/storefrontlib/src/cms-components/myaccount/my-coupons/coupon-card/coupon-card.component.spec.ts

```diff
@@ -25,107 +31,85 @@ class MockUrlPipe implements PipeTransform {
-      .textContent;
-    expect(readMoreLink).toContain('myCoupons.readMore');
-
-    const couponNotificationCheckbox = fixture.debugElement.queryAll(
-      By.css('.form-check-input')
-    );
-    expect(couponNotificationCheckbox.length).toBe(1);
-    const couponNotificationLabel = fixture.debugElement.query(
-      By.css('.form-check-label')
-    ).nativeElement.textContent;
-    expect(couponNotificationLabel).toContain('myCoupons.notification');
-
-    const findProductBtn = fixture.debugElement.query(By.css('button'))
-      .nativeElement.textContent;
-    expect(findProductBtn).toContain('myCoupons.findProducts');
+    expect(el.query(By.css('[data-test]="coupon-name"'))).toBeTruthy();
+    expect(el.query(By.css('[data-test]="coupon-status"'))).toBeTruthy();
+    expect(el.query(By.css('[data-test]="date-start"'))).toBeTruthy();
+    expect(el.query(By.css('[data-test]="date-end"'))).toBeTruthy();
+    expect(el.query(By.css('[data-test]="read-more"'))).toBeTruthy();
+    expect(el.query(By.css('[data-test]="notify-checkbox"'))).toBeTruthy();
+    expect(el.query(By.css('[data-test]="find-button"'))).toBeTruthy();
   });
 
-  it('should be able to open coupon detail dialog', () => {
+  it('should be able to open "read more" dialog', () => {
     fixture.detectChanges();
-    const readMoreLink = fixture.debugElement.query(By.css('a'));
-    readMoreLink.nativeElement.click();
+    el.query(By.css('.cx-card-read-more')).nativeElement.click();
     expect(modalService.open).toHaveBeenCalled();
   });
 
-  it('should be able to subscribe/unsubscribe coupon notification', () => {
-    spyOn(component.notificationChanged, 'emit').and.callThrough();
+  it('should be able to subscribe/unsubscribe notification', () => {
     fixture.detectChanges();
-    const couponNotificationCheckbox = fixture.debugElement.query(
-      By.css('.form-check-input')
-    );
-    couponNotificationCheckbox.nativeElement.click();
+    const chx = el.query(By.css('.form-check-input')).nativeElement;
+    expect(chx.disabled).toEqual(false);
 
-    expect(component.notificationChanged.emit).toHaveBeenCalledWith({
-      notification: true,
-      couponId: 'CustomerCoupon',
-    });
-  });
-
-  it('should be able to show correct notification status', () => {
-    fixture.detectChanges();
-    let couponNotificationCheckbox = fixture.debugElement.queryAll(
-      By.css('.form-check-input')
-    );
-    expect(couponNotificationCheckbox.length).toBe(1);
-    expect(couponNotificationCheckbox[0].nativeElement.checked).toBeFalsy();
-    component.coupon.notificationOn = true;
-    fixture.detectChanges();
-    couponNotificationCheckbox = fixture.debugElement.queryAll(
-      By.css('.form-check-input')
-    );
-    expect(couponNotificationCheckbox.length).toBe(1);
-    expect(couponNotificationCheckbox[0].nativeElement.checked).toBeTruthy();
+    chx.click();
+    expect(component.loading).toEqual(true);
```

Comment:
Is it possible to  expect component.card.notificationChanged.emit.toHaveBeenCalledWith(xxx)?

Reply:
component.loading==true indicates that event has been emitted.